src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs

```diff
@@ -80,34 +80,23 @@ internal class ProjectDependencyFinder
                     projectContext.ProjectFile,
                     framework,
                     preResolvedProjects,
-                    solutionFile,
-                    HoistDependenciesThatAreNotDirectDependencies(projectToResolve, project)
+                    solutionFile
                 );
-                projects.AddRange(dependencies);
```

Comment:
Yes, I meant to. This is the line that caused the breath first traversal of the dependency tree to happen.

The line below adds the direct dependencies, which is all we need now.